events/views.py

```diff
@@ -117,74 +129,160 @@ def __init__(self, **kwargs) -> None:
+                    participant_id=participant
+                    ).exclude(method__in=(ReminderType.EXPIRATION_VOTING_TIME, ReminderType.RUN_ALGORITHM)).first()
+
+                if Event.objects.get(id=meeting_id) not in Event.objects.get_all_meetings():
+                    return redirect('update_event', event_id=meeting_id)
+
+                if OptionalMeetingDates.objects.get_all_event_dates(meeting_id).count() == 0:
+                    return redirect('show_meeting', meeting_id=meeting_id)
+
+                if not participant.is_creator:
+                    return redirect('show_meeting', meeting_id=meeting_id)  # redirect to static meeting page
+
+            except EventParticipant.DoesNotExist:
+                return redirect(HOME_PAGE)
+
+        # check if this is the creator
+        return super(CreateMeetingView, self).dispatch(request, meeting_id, *args, **kwargs)
 
     def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
 
         context = {
+            'title': self.title,
+            'meeting_id': self.meeting_id,
+            'button_text': self.button_text,
+            'reminder_form': self.reminder_form,
             'create_event_form': self.create_event_form,
-            'optional_meetings_formset': self.optional_meetings_formset,
-            'meeting_participants_formset': self.meeting_participants_formset,
             'formset_meeting_data': self.formset_meeting_data,
             'total_meeting_forms': len(self.formset_meeting_data),
             'formset_participant_data': self.formset_participant_data,
+            'optional_meetings_formset': self.optional_meetings_formset,
             'total_participant_forms': len(self.formset_participant_data),
-            'title': 'Create meeting'
+            'meeting_participants_formset': self.meeting_participants_formset,
         }
         return context
 
-    def get(self, request, day=None, month=None, year=None):
-        initial_state = None
+    def get(self, request, meeting_id=None):
 
-        if day and month and year:
-            current_time = datetime.now().time()
+        if meeting_id:
+            meeting_instance = Event.objects.get(id=meeting_id)
+            self.create_event_form = MeetingUpdateForm(instance=meeting_instance)
 
-            initial_state = {
-                'date_time_start': datetime(
-                                        int(year), int(month), int(day), current_time.hour, current_time.minute
-                                    ).strftime("%Y-%m-%dT%H:%M")
-            }
+            self.formset_meeting_data = list(
+                OptionalMeetingDates.objects.get_all_event_dates(meeting_id).exclude(
+                    date_time_start=meeting_instance.date_time_start, date_time_end=meeting_instance.date_time_end
+                    )
+                )
+
+            for index, possible_date in enumerate(self.formset_meeting_data):
```

Comment:
can you please create a different method for lines 193 -198

Reply:
Done 👍